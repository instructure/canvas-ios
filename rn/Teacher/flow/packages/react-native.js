//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

declare var console: typeof console & { disableYellowBox: boolean }

export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}

export type KeyboardEventData = {
  endCoordinates: {
    width: number,
    height: number,
    screenX: number,
    screenY: number,
  },
}
