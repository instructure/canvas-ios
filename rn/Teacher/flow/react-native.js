//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
