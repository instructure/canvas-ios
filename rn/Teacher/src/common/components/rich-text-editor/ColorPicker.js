//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

import React, { Component } from 'react'
import {
  View,
  ScrollView,
  StyleSheet,
  TouchableHighlight,
} from 'react-native'

import i18n from 'format-message'

type Props = {
  pickedColor: (color: string) => void,
}

export function getColors () {
  return {
    '#FFFFFF': i18n('white'),
    '#2D3B45': i18n('black'),
    '#8B969E': i18n('grey'),
    '#EE0612': i18n('red'),
    '#FC5E13': i18n('orange'),
    '#FFC100': i18n('yellow'),
    '#89C540': i18n('green'),
    '#1482C8': i18n('blue'),
    '#65469F': i18n('purple'),
  }
}

export default class ColorPicker extends Component<Props, any> {
  COLORS = getColors()

  render () {
    return (
      <View style={styles.container}>
        <ScrollView horizontal={true} testID={'color-picker.options'}>
          <View style={styles.scrollViewContent}>
            {Object.keys(this.COLORS).map(color => {
              const style = {
                backgroundColor: color,
                borderWidth: color === '#FFFFFF' ? 1 : 0,
                borderColor: color === '#FFFFFF' ? '#E6E9EA' : 'transparent',
              }
              return (
                <TouchableHighlight
                  style={styles.optionContainer}
                  onPress={() => this.props.pickedColor(color)}
                  key={color}
                  testID={`color-picker-option-${color}`}
                  underlayColor='transparent'
                  accessibilityLabel={i18n('Set text color to {colorName} ({hexCode})', {
                    colorName: this.COLORS[color],
                    hexCode: color,
                  })}
                  accessibilityTraits='button'
                >
                  <View style={[styles.option, style]} />
                </TouchableHighlight>
              )
            })}
          </View>
        </ScrollView>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    borderTopWidth: 1,
    borderTopColor: '#C7CDD1',
    backgroundColor: 'white',
  },
  optionContainer: {
    width: 44,
    height: 44,
  },
  option: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    margin: 6,
    borderRadius: 19,
  },
  scrollViewContent: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
})
