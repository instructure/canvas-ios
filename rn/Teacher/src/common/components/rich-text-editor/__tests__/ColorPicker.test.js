//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import ColorPicker from '../ColorPicker'
import explore from '../../../../../test/helpers/explore'

jest
  .mock('react-native/Libraries/Components/Touchable/TouchableHighlight', () => 'TouchableHighlight')

describe('ColorPicker', () => {
  const defaultProps = {
    pickedColor: jest.fn(),
  }

  it('renders', () => {
    expect(
      renderer.create(
        <ColorPicker {...defaultProps} />
      ).toJSON()
    ).toMatchSnapshot()
  })

  it('triggers picked color', () => {
    const pickedColor = jest.fn()
    const tree = renderer.create(
      <ColorPicker {...defaultProps} pickedColor={pickedColor} />
    ).toJSON()

    const white: any = explore(tree).selectByID('color-picker-option-#FFFFFF')
    white.props.onPress()

    expect(pickedColor).toHaveBeenCalledWith('#FFFFFF')

    const black: any = explore(tree).selectByID('color-picker-option-#2D3B45')
    black.props.onPress()

    expect(pickedColor).toHaveBeenCalledWith('#2D3B45')
  })
})
