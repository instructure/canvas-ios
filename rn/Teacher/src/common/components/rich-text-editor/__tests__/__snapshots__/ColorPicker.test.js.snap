// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderTopColor": "#C7CDD1",
      "borderTopWidth": 1,
    }
  }
>
  <RCTScrollView
    horizontal={true}
    testID="color-picker.options"
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <TouchableHighlight
          accessibilityLabel="Set text color to white (#FFFFFF)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#FFFFFF"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#C7CDD1",
                  "borderWidth": 1,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          accessibilityLabel="Set text color to black (#2D3B45)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#2D3B45"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#2D3B45",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          accessibilityLabel="Set text color to grey (#8B969E)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#8B969E"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#8B969E",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          accessibilityLabel="Set text color to red (#EE0612)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#EE0612"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#EE0612",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          accessibilityLabel="Set text color to orange (#FC5E13)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#FC5E13"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#FC5E13",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          accessibilityLabel="Set text color to yellow (#FFC100)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#FFC100"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#FFC100",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          accessibilityLabel="Set text color to green (#89C540)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#89C540"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#89C540",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          accessibilityLabel="Set text color to blue (#1482C8)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#1482C8"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#1482C8",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
        <TouchableHighlight
          accessibilityLabel="Set text color to purple (#65469F)"
          accessibilityTraits="button"
          onPress={[Function]}
          style={
            Object {
              "height": 44,
              "width": 44,
            }
          }
          testID="color-picker-option-#65469F"
          underlayColor="transparent"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 19,
                  "flex": 1,
                  "justifyContent": "center",
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "#65469F",
                  "borderColor": "transparent",
                  "borderWidth": 0,
                },
              ]
            }
          />
        </TouchableHighlight>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
